app/scripts/metamask-controller.js

```diff
@@ -1658,7 +1687,7 @@ export default class MetamaskController extends EventEmitter {
         this.encryptionPublicKeyController.clearUnapproved();
         this.decryptMessageController.clearUnapproved();
         this.signatureController.clearUnapproved();
-        this.approvalController.clear();
+        this.approvalController.clear(ethErrors.provider.userRejectedRequest());
```

Comment:
not sure if this is a necessary change or if this is accidently brought in from rebase conflict resolution

Reply:
this actually fixes a bug that currently exists when the network is switched whiile prrocessing a request.